test/python/ibmq/test_ibmq_qobj.py

```diff
@@ -13,88 +13,101 @@
 
 import os
 import unittest
+import functools
 from qiskit import (ClassicalRegister, QuantumCircuit, QuantumRegister, compile)
-
 from qiskit import IBMQ, Aer
 from qiskit.qasm import pi
-from ..common import requires_qe_access, JobTestCase, slow_test
+
+from ..common import require_multiple_credentials, JobTestCase, slow_test
+
+# Timeout duration
+TIMEOUT = int(os.getenv("IBMQ_TESTS_TIMEOUT", 10))
+BLACKLIST = ('ibmq_4_atlantis',)
+ALWAYS_BLACKLISTED = ('ibmqx_hpc_qasm_simulator',)
 
 
-class TestIBMQQobj(JobTestCase):
-    """Qiskit backend qobj test. Compares remote simulator as
-       configured in environment variables 'IBMQ_QOBJ_DEVICE',
-       'IBMQ_TOKEN' and 'IBMQ_QOBJ_URL' against local simulator
-       'local_qasm_simulator' as ground truth.
+def per_non_blacklisted_backend(*blacklist):
+    """ Test Qobj support on all non-blacklisted backends claiming to support Qobj.
+
+    Args:
+        blacklist (list): List of backend string names to skip.
+
+    Returns:
+        func: Decorator.
     """
+    blacklist = blacklist+ALWAYS_BLACKLISTED
+
+    def per_qobj_backend_decorator(test):
+        @require_multiple_credentials
+        @functools.wraps(test)
+        def _wrapper(self, *args, credentials=[], **kwargs):
+            for qe_token, qe_url in credentials:
+                IBMQ.enable_account(qe_token, qe_url)
+            for backend in IBMQ.backends():
+                config = backend.configuration()
+                if config['allow_q_object'] and backend.name() not in blacklist:
+                    with self.subTest(backend=backend):
+                        backend_test = test if config['simulator'] else slow_test(test)
+                        backend_test(self, backend, *args, **kwargs)
+            for qe_token, _ in credentials:
+                IBMQ.disable_accounts(token=qe_token)
+        return _wrapper
+    return per_qobj_backend_decorator
+
+
+# no blacklist decorator.
+PER_BACKEND = per_non_blacklisted_backend()
+PER_BACKEND_BLACKLISTED = per_non_blacklisted_backend(*BLACKLIST)
```

Comment:
This name is misleading (it seems to mean "for every blacklisted backend"). What about using functions to avoid the capitalized names and have two decorators `@per_qobj_backend` and `@per_restricted_qobj_backend`.